sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/VariantAssignment.java

```diff
@@ -0,0 +1,231 @@
+            return Mono.justOrEmpty(null);
+        }
+
+        Stream<Variant> streamVariant = variants.stream().filter(variant -> {
+            return variant.getName().equals(variantName);
+        }).map(variant -> this.makeVariant(variant));
+
+        return Flux.fromStream(streamVariant).single();
+    }
+
+    @SuppressWarnings("unchecked")
+    private Variant makeVariant(VariantReference variant) {
+        if (variant.getConfigurationValue() != null) {
+            return new Variant(variant.getName(), variant.getConfigurationValue());
+        }
+
+        String reference = variant.getConfigurationReference();
+
+        String[] parts = reference.split("\\.");
+
+        String methodName = "get" + parts[0];
+        Method method = null;
+        Map<String, Object> variantMap = null;
+
+        Optional<VariantProperties> variantProperties = propertiesProvider.stream().filter(properties -> {
+            try {
+                properties.getClass().getMethod(methodName);
+                return true;
+            } catch (NoSuchMethodException | SecurityException e) {
+                return false;
+            }
+        }).findFirst();
+
+        if (!variantProperties.isPresent()) {
+            String message = "Failed to load " + methodName + ". No ConfigurationProperties where found containing it."
+                + ". Make sure it exists and is publicly accessible.";
+            throw new FeatureManagementException(message);
+        }
+
+        // Dynamically Accesses @ConfigurationProperties and finds the matching method.
+        try {
+            method = variantProperties.get().getClass().getMethod(methodName);
+        } catch (NoSuchMethodException | SecurityException e) {
+            String message = "Failed to load " + methodName + " in " + variantProperties.getClass()
+                + ". Make sure it exists and is publicly accessible.";
+            throw new FeatureManagementException(message, e);
+        }
+        // Calls method to get back an Object, this object contains multiple variants
+        // each has a get method.
+        try {
+            variantMap = (Map<String, Object>) method.invoke(variantProperties.get());
+        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+            String message = "Failed invoking " + methodName + " in " + variantProperties.getClass()
+                + ". Make sure it exists and is publicly accessible.";
+            throw new FeatureManagementException(message, e);
+        }
+
+        return new Variant(variant.getName(), variantMap.get(parts[1]));
+    }
+
+    /**
+     * Computes the percentage that the contextId falls into.
+     * 
+     * @param contextId Id of the context being targeted
+     * @return the bucket value of the context id
+     * @throws TargetingException Unable to create hash of target context
+     */
+    private double isTargetedPercentage(String contextId) {
```

Comment:
Seems like duplicated from `TargetingFilter.isTargetedPercentage`

Reply:
Made a shared function, can't removed because it is public api.